Neos.ContentRepository.Core/Classes/NodeType/NodeTypeManager.php

```diff
@@ -198,6 +202,82 @@ public function overrideNodeTypes(array $completeNodeTypeConfiguration): void
         }
     }
 
+    /**
+     * @param NodeType $nodeType
+     * @param NodeName $childNodeName
+     * @return NodeType
+     */
+    public function getTypeOfAutoCreatedChildNode(NodeType $nodeType, NodeName $childNodeName): NodeType
+    {
+        $childNodeTypeName = $nodeType->getNodeTypeNameOfAutoCreatedChildNode($childNodeName);
+        return $this->getNodeType($childNodeTypeName);
+    }
+
+    /**
+     * Return an array with child nodes which should be automatically created
+     *
+     * @return array<string,NodeType> the key of this array is the name of the child, and the value its NodeType.
+     * @api
+     */
+    public function getAutoCreatedChildNodesFor(NodeType $nodeType): array
+    {
+        $childNodeConfiguration = $nodeType->getConfiguration('childNodes');
+        if (!isset($childNodeConfiguration)) {
+            return [];
+        }
+
+        $autoCreatedChildNodes = [];
+        foreach ($childNodeConfiguration as $childNodeName => $configurationForChildNode) {
+            if (isset($configurationForChildNode['type'])) {
+                $autoCreatedChildNodes[NodeName::transliterateFromString($childNodeName)->value]
+                    = $this->getNodeType($configurationForChildNode['type']);
+            }
+        }
+
+        return $autoCreatedChildNodes;
+    }
+
+    /**
+     * Checks if the given $nodeType is allowed as a childNode of the given $childNodeName
+     * (which must be auto-created in $this NodeType).
+     *
+     * Only allowed to be called if $childNodeName is auto-created.
+     *
+     * @param NodeType $parentNodeType The NodeType to check constraints based on.
+     * @param NodeName $childNodeName The name of a configured childNode of this NodeType
+     * @param NodeType $nodeType The NodeType to check constraints for.
+     * @return bool true if the $nodeType is allowed as grandchild node, false otherwise.
+     * @throws \InvalidArgumentException If the given $childNodeName is not configured to be auto-created in $this.
+     */
+    public function allowsGrandchildNodeType(NodeType $parentNodeType, NodeName $childNodeName, NodeType $nodeType): bool
+    {
+        try {
+            $childNodeType = $this->getTypeOfAutoCreatedChildNode($parentNodeType, $childNodeName);
+        } catch (ChildNodeNotConfigured $exception) {
+            throw new \InvalidArgumentException(
+                'The method "allowsGrandchildNodeType" can only be used on auto-created childNodes, '
+                . 'given $childNodeName "' . $childNodeName->value . '" is not auto-created.',
+                1403858395,
+                $exception
+            );
```

Comment:
this is error is currently thrown when creating new nodes in the ui

<img width="662" alt="image" src="https://github.com/neos/neos-development-collection/assets/85400359/6cc28803-7ff9-4ac2-92ea-88182bf81066">

Reply:
Fixed via https://github.com/neos/neos-development-collection/pull/4520/commits/22b41cf5d804553c018137d0b77a461123402478